### PR TITLE
[ckpt] fix: fix resume

### PR DESCRIPTION
**Pull request by FoolPlayer**
### Setup of the bug

```
┌──────────────────────────────────────────────────────────────────────────┐
│ Trained for 40 steps, but save_steps=5, so the last save is at step 35. │
│ Steps 36–40 were never saved before Ctrl-C.                             │
│ ──────────────────────────────────────────                              │
│ Latest ckpt on disk = global_step_35   ← this is what resume loads      │
└──────────────────────────────────────────────────────────────────────────┘
```

### The cascade

```
RESUME → load ckpt(35)
        state.global_step = 35
        start_epoch = 35 // 30 = 1
        start_step  = 35 % 30  = 5

  enter epoch 1
  └── for _ in range(5, 30):  train_step()
       │
       │  ┌─────────────────────────────────────────────────────────┐
       │  │  train_step():                                          │
       │  │                                                         │
       │  │    state.global_step += 1     # 35 → 36   ← Bug 2       │
       │  │           │                                             │
       │  │           ▼                                             │
       │  │    next(data_iterator)        # immediate StopIteration │
       │  │           │                   #   ← Bug 1               │
       │  │           │                   # (StatefulDataLoader     │
       │  │           │                   #  state never restored)  │
       │  │           ▼                                             │
       │  │    [except StopIteration → break]                       │
       │  └─────────────────────────────────────────────────────────┘
       ▼
  on_epoch_end(state):
       global_step = 36
       _last_saved_step = -1     ← callback state is not persisted,
                                   so a fresh process resets it to -1
       check:  36 ≠ -1  →  SAVE                              ← Bug 3
                  │
                  ▼
              ckpt(36)  ← bogus: this "step" never actually trained
       │
       ▼  epoch 1 finishes
  start_step is reset to 0, training advances to epoch 2
  (UI shows "Epoch 3/3")
       └── 30 fresh steps run normally — but the progress bar shows
           "0/30", which is what made it look like an epoch was skipped.
```

### After the fix

```
  train_step():
      next(data_iterator)        # fetch first
              │
              ▼  StopIteration → break  (global_step stays at 35)
                                  ↓
                                state.global_step += 1   ← never reached

  on_epoch_end(state):
      global_step (35) ≤ _epoch_start_global_step (35)
      → no step ran this epoch, skip save                  ← Bug 3 fixed

  Next resume still loads ckpt(35); no spurious ckpt(36) to mislead it.
```

### Three bugs, one chain reaction

| Bug | Location | Symptom |
|-----|----------|---------|
| 1 | `StatefulDataLoader` state was not actually restored | the very first `next()` raises `StopIteration` |
| 2 | `state.global_step += 1` happened **before** `next()` | step counter advances even though no batch was processed |
| 3 | `on_epoch_end` guard only checks `_last_saved_step` (reset to `-1` after a restart) | a "ghost" checkpoint at the inflated step number gets written |

Only **Bug 1** is the real root cause (dataloader resume). Bugs **2** and **3** amplified the symptom into "phantom checkpoint + skipped epoch". This patch only suppresses Bugs **2** and **3** so the symptoms go away; Bug **1** is tracked separately.
### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

File: veomni/trainer/callbacks/checkpoint_callback.py [hunk 1]
```diff
@@ -47,7 +48,22 @@ def on_step_end(self, state: TrainerState, **kwargs):
         if self.every_n_steps and state.global_step % self.every_n_steps == 0:
             self._save_checkpoint(state)
 
+    def on_epoch_begin(self, state: TrainerState, **kwargs) -> None:
+        self._epoch_start_global_step = state.global_step
```
Comment (Coach257) [inline veomni/trainer/callbacks/checkpoint_callback.py:52]: It seems `self._last_saved_step = state.global_step` also skips saving no training ckpt.
